sensor/kubernetes/listener/resources/networkpolicy_store.go

```diff
@@ -194,3 +194,20 @@ func (n *networkPolicyStoreImpl) Find(namespace string, podLabels map[string]str
 	}
 	return results
 }
+
+// OnNamespaceDeleted reacts to a namespace deletion, deleting all network policies in this namespace from the store.
+func (n *networkPolicyStoreImpl) OnNamespaceDeleted(namespace string) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	defer n.updateStateMetric()
+
+	netpols, found := n.data[namespace]
+	if !found {
+		return
+	}
+
+	for id := range netpols {
+		delete(n.data[namespace], id)
+	}
+	delete(n.data, namespace)
```

Comment:
nit: Shouldn't just this be enough to delete the whole thing? I assume that by using map's `delete` method you're anyway just dereferencing and the memory will be freed by garbage collection at some point. So maybe deleting the top-level reference would do the trick?

Reply:
To be honest I'm not sure how the garbage collector works in go. If we are sure it will delete the inner objects, I'm fine with deleting just the top-level reference.

Reply:
I think keeping only `delete(n.data, namespace)` should work too, but deletion of namespaces is rare, so even if there is a penalty for iterating, it would not have any influence. I am fine, keeping it as it is.